N8N-3750 Redesign button component

alexgrozav opened the pull request.

alexgrozav committed:
feat: Design system color improvements and button component redesign.

alexgrozav committed:
feat: Added button focus state and unit tests.

alexgrozav committed:
refactor: Aligned n8n-button usage inside of editor-ui.

alexgrozav committed:
test: Updated snapshots.

alexgrozav committed:
refactor: Extracted focus outline width into scss variable.

alexgrozav committed:
fix: Fixed select input border-radius.

alexgrozav committed:
refactor: Removed element-ui references in button.

alexgrozav committed:
fix: Fixed scss variable imports.

alexgrozav committed:
feat: Added color-neutral variable story.

alexgrozav committed:
fix: Fixed color-secondary variable definition.

alexgrozav committed:
feat: Added color-white story.

alexgrozav committed:
test: Updated button snapshot.

alexgrozav committed:
feat: Replaced zoom buttons with new n8n-icon-button.

alexgrozav committed:
feat: Added stories for float utilities.

alexgrozav committed:
chore: Updated color shades generation code for later use.

alexgrozav committed:
chore: Removed color-white code.

alexgrozav committed:
chore: Updated story properties for button components.

alexgrozav committed:
fix: Added el-button fallback for places where el-button is not replaceable (messagebox).

alexgrozav committed:
chore: Merged master into n8n-3750-redesign-button-component

alexgrozav committed:
feat: Reverted to css modules. Replaced el-button with n8n-button at application level.

alexgrozav committed:
test: Updated button snapshot.

alexgrozav committed:
fix: Fixed element-ui locally referenced buttons (via components: {}).

mutdmour commented on packages/design-system/src/components/N8nIconButton/IconButton.vue:
```diff
@@ -23,13 +13,15 @@ export default {
-			type: String,
+			default: 'primary',
+			validator: (value: string): boolean =>
+				['primary', 'secondary', 'tertiary', 'success', 'warning', 'danger'].includes(value),
 		},
 		size: {
 			type: String,
 			default: 'medium',
+			validator: (value: string): boolean =>
+				['mini', 'small', 'medium', 'large', 'xlarge'].includes(value),
```
would recommend removing validation here.. that way `n8n-button` component can handle validation. and we will get one warning instead of 2 warnings.. also don't have to update this one if we change that

alexgrozav replied:
Good suggestion! Removed.

mutdmour commented on packages/design-system/theme/src/_tokens.scss:
```diff
@@ -1,405 +1,448 @@
-	--spacing-3xl: 4rem;
-	--spacing-4xl: 8rem;
-	--spacing-5xl: 16rem;
+  --color-primary-h: 7;
+  --color-primary-s: 100%;
+  --color-primary-l: 68%;
+  --color-primary: hsl(
```
it looks like you changed a few different colors here in this file... was this intentional? can you double check with Max? Pretty sure the floats make a difference..

alexgrozav replied:
It resolves to the same hex code, but reverted the changes. Thanks!

mutdmour commented on packages/editor-ui/vue.config.js:
```diff
@@ -27,6 +28,12 @@ module.exports = {
 		],
+		resolve: {
+			alias: {
+				'element-ui/packages/button': path.resolve(__dirname, '..', 'design-system/src/components/N8nButton/Button.vue'),
+				'element-ui/lib/button': path.resolve(__dirname, '..', 'design-system/src/components/N8nButton/Button.vue'),
```
nice, though I expected this to link to a wrapper component... that maps element ui props to our own props... that way if they define size `mini` we can translate that to our size `2xsmall` for example.. also won't break how it's used by other element ui elements. though button might be too simple for this..

mutdmour commented on packages/editor-ui/src/components/NodeExecuteButton.vue:
```diff
@@ -4,6 +4,7 @@
 		:type="type"
+		:outline="outline"
 		:size="size"
 		:transparentBackground="transparent"
```
guess this needs to be deleted

mutdmour commented on packages/editor-ui/src/components/FixedCollectionParameter.vue:
```diff
@@ -287,6 +287,13 @@ export default mixins(genericHelpers)
 </script>
 
 <style scoped lang="scss">
+::v-deep {
+	.button {
+		--button-background-color: var(--color-background-base);
+		--button-border-color: var(--color-foreground-base);
+	}
+}
```
we should try to avoid adding overrides as much as possible. why are these needed when it was not needed before? can we not add a variation of this theme that works? (applies to the two other overrides in this PR)
if current impl requires this, talk to Max, he can recommend a way to use button without overrides.

alexgrozav replied:
Unfortunately, we've got a special case here where the tertiary button must have a different background. I've discussed it with Max and there's a planned replacement for the way we visualize the fields in NDV and we're ok to compromise for now.

alexgrozav committed:
fix: Updated colors. Removed irrelevant validation. Added ElButton override component.

alexgrozav committed:
test: Updated button override snapshot.

alexgrozav committed:
fix: Various button adjustments and fixes.

alexgrozav committed:
chore: Merged master into n8n-3750-redesign-button-component-2.

alexgrozav committed:
chore: Merge master into n8n-3750-redesign-button-component-2.

alexgrozav committed:
fix: Updated button disabled state.

alexgrozav committed:
test: Updated snapshots.

alexgrozav committed:
Merge branch 'master' into n8n-3750-redesign-button-component-2

mutdmour commented on packages/design-system/theme/src/_tokens.scss:
```diff
@@ -1,405 +1,448 @@
+
+  --color-primary-tint-2-h: 11;
+  --color-primary-tint-2-s: 100%;
+  --color-primary-tint-2-l: 95%;
```
can you make sure to revert all values that changed?

alexgrozav replied:
Yes, sorry. I updated the values to reflect what's in Figma at first, but spoke to Max about it afterwards and found out they're not a 100% match to make theming easier. Updated.

alexgrozav committed:
fix: Consolidated css variables changes.

alexgrozav committed:
Merge branch 'master' into n8n-3750-redesign-button-component-2

alexgrozav committed:
chore: Merged branch master into n8n-3750-redesign-button-component-2.

alexgrozav committed:
Data pinning (#3512)

* refactor: Aligned n8n-button usage inside of editor-ui.

* feat: Added edit data button on json hover.

* feat: Extracted code editor into separate form component.

* feat: Added edit data button on json hover.

* feat: Added pinData and edit mode methods.

* :fire: Remove conflict markers

* :pencil2: Update i18n keys

* :zap: Add JSON validation

* :card_file_box: Add `pinData` column to `workflow_entity`

* :blue_book: Tighten type

* :zap: Make `pinData` column nullable

* :zap: Adjust workflow endpoints for pin data

* :blue_book: Improve types

* :pencil2: Improve wording

* Inject pindata into items flow (#3420)

* :zap: Inject pin data - Second approach

* :fire: Remove unneeded lint exception

* feat: Added edit data button on json hover.

* feat: Extracted code editor into separate form component.

* feat: Added edit data button on json hover.

* fix: Fixed rebase conflicts.

* :rewind: Undo button change

* :bug: Fix runNode call

Adjust per update in bdb84130d687811d65337ff6b025e7cb0eae8256

* :test_tube: Fix workflow tests

* :bug: More merge conflict fixes

* feat: Added pin/unpin button and store mutations.

* feat: Size check. Various design and ux improvements.

* :zap: Add transformer

* :zap: Hoist pin data

* :zap: Adjust endpoints for hoisted pin data

* :blue_book: Expand interface

* :bug: Fix stray array

* :shirt: Fix build

* :shirt: Add lint exception

* :shirt: Fix header

* :art: Add color secondary tints

* :sparkles: Create `HeaderMessage` component

* :zap: Adjust `InfoTip` component

* :sparkles: Add `HeaderMessage` to `RunData`

* :bug: Fix console error

* :shirt: Fix lint

* :zap: Consolidate `HeaderMessage` and `Callout`

* :rewind: Undo `InfoTip` changes

* :fire: Remove duplicate icons

* :zap: Simplify template

* :art: Change cursor for action text

* :shirt: Fix lint

* :zap: Add URL

* :bug: Fix handler name

* :zap: Use constant

* :recycle: Refactor per feedback

* fix: Various fixes after data pinning relocation.

* fix: Added store mutation for setting pinned data.

* feat: Added pinned state for workflow canvas node.

* fix: Fixed workflow saving.

* fix: Removed pinData hoisting (no longer necessary).

* feat: Added canPinData flag to hide for input pane and binary data. Fixed unpin and execute flow.

* :zap: Fixes for canvas pin data (#3587)

* :zap: Fixes for canvas pin data

* :blue_book: Rename type

* :test_tube: Fix unrelated Public API test

* :fire: Remove logging

* feat: Updated pinData mixin to no longer include extra fields.

* :zap: Output same pindata for every run

* :art: Fix cropping

* :fire: Remove unrelated logging

* feat: Moved edit button next to pin button.

* feat: Changed data to be inserted for empty state.

* chore: Changed invalid editor output translation.

* feat: Added error line reporting on JSON Validation.

* feat: Migrated pinData edit mode to store.

* chore: Merged duplicate node border color condition.

* feat: Moved pin data validation to mixin. Added check before closing ndv modal.

* fix: Changed pinned data size calculation to discard active node pin data.

* feat: Added support for rename and delete node with pin data.

* feat: Simplified editing state. Fixed edit mode in input panel after store migration.

* feat: Various data pinning improvements.

* fix: Fixed callout link underline.

* refactor: Added support for both string and objects for data size check.

* feat: Added disabled node check for input panel. Fixed monaco editor resizing.

* fix: Fixed edit mode footer size.

* :zap: Fix pindata items per run

* :shirt: Remove unneeded exception

* refactor: Added isValidPinData() helper method.

* refactor: Changed how string size in bytes in calculated.g

* refactor: Updated pinData mixin interface.

* refactor: Merged filter and reduce in pinDataSize calculation.

* fix: Changed code-editor to correct type.

* fix: Added insert test data message to trigger nodes.

* feat: Disabled data pinning for multiple output nodes.

* refactor: Updated ndv.input.disabled translation to include node name.

* refactor: Aligned n8n-button usage inside of editor-ui.

* feat: Added edit data button on json hover.

* feat: Extracted code editor into separate form component.

* feat: Added edit data button on json hover.

* feat: Added pinData and edit mode methods.

* :fire: Remove conflict markers

* :pencil2: Update i18n keys

* :zap: Add JSON validation

* :card_file_box: Add `pinData` column to `workflow_entity`

* :blue_book: Tighten type

* :zap: Make `pinData` column nullable

* :zap: Adjust workflow endpoints for pin data

* :blue_book: Improve types

* :pencil2: Improve wording

* Inject pindata into items flow (#3420)

* :zap: Inject pin data - Second approach

* :fire: Remove unneeded lint exception

* feat: Added edit data button on json hover.

* feat: Extracted code editor into separate form component.

* feat: Added edit data button on json hover.

* fix: Fixed rebase conflicts.

* :rewind: Undo button change

* :bug: Fix runNode call

Adjust per update in bdb84130d687811d65337ff6b025e7cb0eae8256

* :test_tube: Fix workflow tests

* :bug: More merge conflict fixes

* feat: Added pin/unpin button and store mutations.

* feat: Size check. Various design and ux improvements.

* :zap: Add transformer

* :zap: Hoist pin data

* :zap: Adjust endpoints for hoisted pin data

* :blue_book: Expand interface

* :bug: Fix stray array

* :shirt: Fix build

* :art: Add color secondary tints

* :sparkles: Create `HeaderMessage` component

* :zap: Adjust `InfoTip` component

* :sparkles: Add `HeaderMessage` to `RunData`

* :bug: Fix console error

* :shirt: Fix lint

* :zap: Consolidate `HeaderMessage` and `Callout`

* :rewind: Undo `InfoTip` changes

* :fire: Remove duplicate icons

* :zap: Simplify template

* :art: Change cursor for action text

* :shirt: Fix lint

* :zap: Add URL

* :bug: Fix handler name

* :zap: Use constant

* :recycle: Refactor per feedback

* fix: Various fixes after data pinning relocation.

* fix: Added store mutation for setting pinned data.

* feat: Added pinned state for workflow canvas node.

* :zap: Fixes for canvas pin data (#3587)

* :zap: Fixes for canvas pin data

* :blue_book: Rename type

* :test_tube: Fix unrelated Public API test

* :fire: Remove logging

* feat: Updated pinData mixin to no longer include extra fields.

* fix: Removed pinData hoisting (no longer necessary).

* chore: Merged duplicate node border color condition.

* :zap: Output same pindata for every run

* :art: Fix cropping

* :bug: Fix excess closing template tag

* fix: Removed rogue template tag after merge.

* fix: Fixed code-editor resizing when moving ndv panel.

* feat: Added node duplication pin data.

* :zap: Implement telemetry

* :recycle: Add clarifications from call

* fix: Fixed run data header height.

* feat: Removed border from pin data callout.

* feat: Added line-break before 'or insert pin data'.

* feat: Changed enterEditMode to always insert test data if there's no execution data.

* feat: Removed copy output tooltip.

* feat: Removed unpin tooltip.

* fix: Removed thumbtack icon rotation.

* fix: Removed run info from Edit Output title.

* feat: Hid edit and pin buttons when editing.

* feat: Updated monaco code-editor padding and borders.

* feat: Progress on pinData error message format

* feat: Updated copy feature to work without any selected value.

* feat: Moved save and cancel buttons. Cleared notifications on save.

* feat: Changed pin data beforeClosing confirm text.

* feat: Closing ndv when discarding or saving pindata on close.

* feat: Added split in batches node to pin data denylist.

* fix: Added missing margin-bottom to webhook node.

* feat: Moved thumbtack icon to the right, replacing the checkmark.

* fix: Hid pagination while editing.

* feat: Added pin data discovery flow.

* feat: Changed pin data discovery flow to avoid tooltip glitching.

* fix: Changed copy selection to copy all input data.

* feat: Updated pin data validation error message for unexpected single quotes.

* fix: Replaced :manual='true' prop with manual shorthand.

* fix: Removed unused variable.

* chore: Renamed translation key to node.discovery.pinData.

* refactor: Extracted isPinDataNodeType to pinData mixin.

* fix: Updated watch condition to improve performance.

* refactor: Renamed some pin data variables and methods as per review.

* fix: Added partial translation for JSON.parse pin data error messages.

* chore: Temporarily disabled failing unit test.

* :test_tube: Fix data pinning workflow retrieval test

* :fire: Remove unused imports

* :fire: Remove leftover line

* :zap: Skip pindata node issues on BE

* :zap: Skip pindata node issues on FE

* :zap: Hide `RunInfo` for pindata node

* :zap: Hide purple banner in edit output mode

* feat: Updated data pinning discoverability flow.

* fix: Fixed paginated data pinning.

* fix: Disabled pin data in read only mode.

* :bug: Fix runtime error with non-array

* fix: Loading pin data when opening execution.

* :zap: Adjust stale data warning for pinned data

* :zap: Skip auth in endpoint

* :zap: Mark start node for pinned trigger

* :pencil2: Comment on passthrough

* :fire: Remove comment

* Final pindata metrics changes (#3673)

* :bug: Fix `pinData` tracked as `0`

* :zap: Add `is_pinned` to `nodesGraph`

* :blue_book: Extend `IWorkflowBase`

* :zap: Handle `pinData` being `undefined`

* :zap: Add `data_pinning_tooltip_presented`

* :recycle: Refactor to remove circular dependency

* fix: Added pin data handling when importing workflow. (#3698)

* :fire: Remove helper from WorkflowExecute

* :zap: Add logic for single pinned trigger

* :shirt: Remove lint exception

* fix: Added pin data handling in importWorkflowExact.

* N8N-4077 data pinning discoverability part 2 (#3701)

* fix: Fixed pin data discovery tooltip position when moving canvas.

* feat: Updated data pinning discovery tooltip copy.

* Fix data pinning build (#3702)

* :zap: Disable edit button for disabled node

* :zap: Ensure disabled pinned nodes are passthrough

* :bug: Fix JSON key unfurling in edit mode

* :zap: Improve implementation

* :bug: Fix console error

* fix: Fixed copying pinned output data. (#3715)

* Fix pinning for webhook responding with output from last node (#3719)

* fix: Fixed entering edit mode after refresh.

* fix: Fixed type error during build.

* fix: RunData import formatting.

* chore: Updated pin data types.

* fix: Added missing type to stringSizeInBytes.

Co-authored-by: Iván Ovejero <ivov.src@gmail.com>

alexgrozav committed:
fix: Showing pin data without executing the node only in output pane.

alexgrozav committed:
fix: Updated no data message when previous node not executed.

alexgrozav committed:
feat: Added expression input and evaluation for pin data nodes without execution.

alexgrozav committed:
chore: Fixed linting issues and removed remnant console.log().

alexgrozav committed:
chore: Undone package-lock changes.

alexgrozav committed:
fix: Removed pin data store changes.

alexgrozav committed:
fix: Created a new object using vuex runExecutionData.

alexgrozav committed:
fix: Fixed bug appearing when adding a new node after executing.

ivov committed:
fix: Fix editor-ui build

alexgrozav committed:
feat: Added green node connectors when having pin data output.

alexgrozav committed:
Merge branch 'n8n-4143-make-node-parameter-variables-and' of github.com:n8n-io/n8n into n8n-4143-make-node-parameter-variables-and

alexgrozav committed:
chore: Fixed linting errors.

alexgrozav committed:
fix: Added pin data eventBus unsubscribe.

alexgrozav committed:
fix: Added pin data color check after adding a connection.

alexgrozav committed:
Merge pull request #3738 from n8n-io/n8n-4143-make-node-parameter-variables-and

N8N-4143 Variables and expressions for pin data nodes without execution

ivov committed:
:twisted_rightwards_arrows: Merge master

ivov committed:
:art: Add pindata styles

janober commented:
Got released with `n8n@0.187.1`
